frontend/src/actions/userActions.js

```diff
@@ -194,7 +194,7 @@ export const listFollowing = () => async (dispatch) => {
 export const updateUserProfile = (userData) => async (dispatch, getState) => {
   try {
     let { user } = await usersService.updateUserProfile(userData);
-    user.profile.email = user.email;
+    user.profile.email = user.email || userData.email;
```

Comment:
This is not actually a solutions. Sure this will update UI. But what if this actions has failed. Can you handle that?

Reply:
If the action fails, nothing will happen and there will no change in UI